commands/builds/create.js

```diff
@@ -43,25 +43,36 @@ function compressSource (context, dir, tempFile) {
   })
 }
 
-async function uploadDirToSource (context, heroku, dir) {
-  let tempFilePath = path.join(os.tmpdir(), uuid.v4() + '.tar.gz')
+async function uploadDirToSource (context, heroku, tarPath) {
+  let filePath
+
+  if (fs.statSync(tarPath).isDirectory()) {
+    filePath = path.join(os.tmpdir(), uuid.v4() + '.tar.gz')
+    await compressSource(context, tarPath, filePath)
+  } else {
+    filePath = tarPath
+  }
+
   let source = await heroku.post('/sources')
-  await compressSource(context, dir, tempFilePath)
+
   await cli.got.put(source.source_blob.put_url, {
-    body: fs.createReadStream(tempFilePath),
+    body: fs.createReadStream(filePath),
     headers: {
       'Content-Type': '',
-      'Content-Length': fs.statSync(tempFilePath).size
+      'Content-Length': fs.statSync(filePath).size
     }
   })
   return source.source_blob.get_url
 }
 
 async function create (context, heroku) {
   let sourceUrl = context.flags['source-url']
+  let sourceTar = context.flags['source-tar']
   let dir = context.flags.dir || process.cwd()
 
-  if (!sourceUrl) {
+  if (sourceTar) {
+    sourceUrl = await uploadDirToSource(context, heroku, path.join(dir, sourceTar))
```

Comment:
What if we passed an absolute path here?

Reply:
makes sense, updating ...